### PR TITLE
StarUML: 2.8.1 -> 4.0.1

### PR DESCRIPTION
**Pull request by noetarbouriech**
###### Motivation for this change
Update to latest version 4.0.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/staruml/default.nix [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-, libXdamage, expat }:
+, gtk3, glib, systemd
+, xorg, nss, nspr
+, atk, at-spi2-atk, dbus
+, gdk-pixbuf, pango, cairo
+, expat, libdrm, mesa
+, alsa-lib, at-spi2-core, cups }:
 
 let
   LD_LIBRARY_PATH = lib.makeLibraryPath
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:11]:
```suggestion
  LD_LIBRARY_PATH = lib.makeLibraryPath [
```

File: pkgs/tools/misc/staruml/default.nix [hunk 2]
```diff
@@ -1,23 +1,32 @@
...
-    [ glib gtk2 gdk-pixbuf alsa-lib nss nspr GConf cups libgcrypt dbus libXdamage expat ];
+  [ glib gtk3 xorg.libXdamage
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:12]:
```suggestion
  glib gtk3 xorg.libXdamage
```

File: pkgs/tools/misc/staruml/default.nix [hunk 3]
```diff
@@ -1,23 +1,32 @@
...
+  xorg.libX11 xorg.libxcb xorg.libXcomposite
+  xorg.libXcursor xorg.libXext xorg.libXfixes
+  xorg.libXi xorg.libXrender xorg.libXtst
+  nss nspr atk at-spi2-atk dbus
+  gdk-pixbuf pango cairo
+  xorg.libXrandr expat libdrm
+  mesa alsa-lib at-spi2-core
+  cups ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:20]:
```suggestion
  cups
  ];
```

File: pkgs/tools/misc/staruml/default.nix [hunk 4]
```diff
@@ -30,25 +39,23 @@ stdenv.mkDerivation rec {
...
     ln -s ${lib.getLib systemd}/lib/libudev.so.1 $out/lib/libudev.so.0
 
-    for binary in StarUML Brackets-node; do
-      ${patchelf}/bin/patchelf \
-        --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
-        $out/bin/$binary
-      wrapProgram $out/bin/$binary \
-        --prefix LD_LIBRARY_PATH : $out/lib:${LD_LIBRARY_PATH}
-    done
+    ${patchelf}/bin/patchelf \
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:48]:
```suggestion
    patchelf \
```
and add patchelf to nativeBuildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Review by noetarbouriech — COMMENTED** [review]
Changes pushed

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135599` at ca71c064 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>staruml</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/misc/staruml/default.nix:56:3:
```
   |
56 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/staruml/default.nix:41:3:
```
   |
41 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `patchelf`.
Near pkgs/tools/misc/staruml/default.nix:2:9:
```
  |
2 | , dpkg, patchelf
  |         ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/staruml/default.nix [hunk 5]
```diff
@@ -1,23 +1,33 @@
...
+  LD_LIBRARY_PATH = lib.makeLibraryPath [
+  glib gtk3 xorg.libXdamage
+  xorg.libX11 xorg.libxcb xorg.libXcomposite
+  xorg.libXcursor xorg.libXext xorg.libXfixes
+  xorg.libXi xorg.libXrender xorg.libXtst
+  nss nspr atk at-spi2-atk dbus
+  gdk-pixbuf pango cairo
+  xorg.libXrandr expat libdrm
+  mesa alsa-lib at-spi2-core
+  cups
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:20]:
```suggestion
    glib gtk3 xorg.libXdamage
    xorg.libX11 xorg.libxcb xorg.libXcomposite
    xorg.libXcursor xorg.libXext xorg.libXfixes
    xorg.libXi xorg.libXrender xorg.libXtst
    nss nspr atk at-spi2-atk dbus
    gdk-pixbuf pango cairo
    xorg.libXrandr expat libdrm
    mesa alsa-lib at-spi2-core
    cups
```

File: pkgs/tools/misc/staruml/default.nix [hunk 6]
```diff
@@ -30,25 +40,23 @@ stdenv.mkDerivation rec {
...
+      --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+      $out/bin/staruml
+    wrapProgram $out/bin/staruml \
+      --prefix LD_LIBRARY_PATH : $out/lib:${LD_LIBRARY_PATH}
   '';
 
   meta = with lib; {
     description = "A sophisticated software modeler";
     homepage = "https://staruml.io/";
     license = licenses.unfree;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/staruml/default.nix:59]:
```suggestion
    license = licenses.unfree;
    maintainers = with maintainers; [ ];
```

**Pull request merged**